### ISSUE TITLE
Automate hyperparameter optimisation

### ISSUE THREAD

**Issue by charleygros** [ivadomed/ivadomed#75]
We were discussing with @olix86 about hyperparameter optimisation to evaluate our recent new features. Quickly felt overwhelmed by the number of tests that would be needed to be run.

What we discussed:
- Fix a GH commit for all experiments.
- Ask @alexfoias not to add new data during the experiments.
- Save the paths of the subjects used for these experiments.
- Fix a `random_seed`, `dropout`, `batch_norm_momentum`, `num_epochs`.
- Fix common transformations.
- Create a script to loop across the experiments --> at each iteration, change the `log_directory` + update hyperparameter value.
- Not to use the `testing` dataset at this stage (hyperparameter optimisation), only training and validation on TensorBoard.

A first batch of experiments could be:
- Model: Unet
- Dataset: lesion
- Loss: Dice Loss
- Hyperparameters:
    (i) batch size: 8, 16, 32, 64
    (ii) initial learning rate: 1e-3, 1e-4, 1e-5
    (iii) LR scheduler:
        (a) CyclicLR: XX, XX
        (b) CosineAnnealingWarm: XX, XX
